DevelopmentProposals/readme.md

```diff
@@ -0,0 +1,12 @@
+## Overview
+
+This directory contains all the development proposals for OpenSim.
+
+## How to create a proposal
+
+1. Create a new directory under directory ***DevelopmentProposals***.
```

Comment:
> I think it was mentioned that Sanbox is going to move to top-level directory in future.

I did mention this.

I'm fairly indifferent about where this directory goes. In the long term, I'd prefer for the `OpenSim` directory to contain only the library source code; I think `OpenSim/Examples` and `OpenSim/Tests` should be moved to the root of the repo.